### PR TITLE
Update Dockerfile to copy the project into the image

### LINKED ISSUE #2076: Dockerfile template is not working
**Describe the bug**
The current dockerfile template is not working, as it does not copy the serverpod project into the container.

**To Reproduce**
1. `serverpod create example`
2. `docker build .` inside the example server
3. "No pubspec.yaml found"

**Expected outcome**
Docker image can be built from Dockerfile

**Serverpod versions**
 -  _CLI Version:_ 1.2.5
 - _Library version:_ 1.2.5

**Platform information**
Macos Arm

### PR DESCRIPTION
**Pull request by dkbast**
Fixes #2076

## Pre-launch Checklist

- [x] I read the [Contribute](https://docs.serverpod.dev/contribute) page and followed the process outlined there for submitting PRs.
- [x] This update contains only one single feature or bug fix and nothing else. (If you are submitting multiple fixes, please make multiple PRs.)
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`), and made sure that the documentation follows the same style as other Serverpod documentation. I checked spelling and grammar.

### CODE REVIEW

**Comment by dkbast** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by inf0rmatix** [comment]
I have read the CLA Document and I hereby sign the CLA

File: templates/serverpod_templates/projectname_server_upgrade/Dockerfile [hunk 1]
```diff
@@ -21,9 +21,10 @@ COPY --from=build /runtime/ /
...
 COPY --from=build /app/config/ config/
 COPY --from=build /app/web/ web/
+COPY --from=build /app/migrations/ migrations/
 
 EXPOSE 8080
 EXPOSE 8081
 EXPOSE 8082
 
-CMD app/bin/main --mode $runmode --server-id $serverid --logging $logging --role $role
+CMD app/bin/main --mode $runmode --server-id $serverid --logging $logging --role $role --apply-migrations
```
Comment (SandPod) [inline templates/serverpod_templates/projectname_server_upgrade/Dockerfile:30]: Question: Applying migrations is often a step that the developer wants to have control over. Imagine starting multiple containers in one go, then we probably don't want all of them to apply the migrations.

Therefore we probably don't want them to always be applied but make it possible to specify them if needed.

I think we could utilize a docker [ENTRYPOINT](https://docs.docker.com/reference/dockerfile/#entrypoint) to achieve this. Where a resonable default is provided as the `CMD` but all flags can be overridden from the ENTRYPOINT.

File: templates/serverpod_templates/projectname_server_upgrade/Dockerfile [hunk 2]
```diff
@@ -3,7 +3,7 @@
 FROM dart:3.2.5 AS build
 
 WORKDIR /app
-COPY . .
```
Comment (SandPod) [inline templates/serverpod_templates/projectname_server_upgrade/Dockerfile:6]: Question: I don't understand the motivation behind this change.

I can successfully build the image in the `build` container when testing.

The above `WORKDIR` command sets the path inside of the build container to `/app`. So `COPY . .` __should__ provide the same result as `COPY . /app`.

Comment (SandPod) [reply templates/serverpod_templates/projectname_server_upgrade/Dockerfile:6]: Note that we are not copying the full project from the build container into the final container that runs the server. We only copy the files/folders listed on line 20-24.

**Review by SandPod — CHANGES_REQUESTED** [review]
Thanks for taking this on! 🙌 

I have a couple of questions and concerns.

File: templates/serverpod_templates/projectname_server_upgrade/Dockerfile (hunk 2 above)
Comment (inf0rmatix) [reply templates/serverpod_templates/projectname_server_upgrade/Dockerfile:6]: This works on Linux but not with Docker for Mac

Comment (dkbast) [reply templates/serverpod_templates/projectname_server_upgrade/Dockerfile]: This seems to have been an issue with docker desktop for mac, but I cannot reproduce it anymore, so Ill revert it :)

**Comment by SandPod** [comment]
I think the issue this PR is trying to fix (a part from copying over the migration files) will be solved here: https://github.com/serverpod/serverpod/pull/2193

**Comment by dkbast** [comment]
> I think the issue this PR is trying to fix (a part from copying over the migration files) will be solved here: #2193

Yes I think that would be great - we are currently focussing on the WIP PR #2124 which is based on this change, but we are mostly adding a docker based deploy strategy to the templates

**Review by BenAuerDev — COMMENTED** [review]
@dkbast hey man do you create a password.yaml from a base64/github secrets? 

And if so, where are you copying it over to the vps?

**Comment by dkbast** [comment]
> @dkbast hey man do you create a password.yaml from a base64/github secrets?
> 
> And if so, where are you copying it over to the vps?

I decided to go with plaintext secrets as this is how the other two workflows were already setup. We don't explicitly copy over the passwords file, but we build the serverpod binary, package it up in a docker image and ship that.

**Comment by BenAuerDev** [comment]
Ah 😊 sounds cool. 

Maybe I should look into that aswell 💪

**Comment by dkbast** [comment]
this is already done in #2193

**Pull request closed without merging**